string name
float32 turn
